tests/subsys/net/lib/nrf_cloud/fota_common/src/main.c

```diff
@@ -0,0 +1,546 @@
+
+	int ret = nrf_cloud_bootloader_fota_slot_set(&job);
+
+	zassert_equal(-ENOTSUP, ret,
+		      "return should be -ENOTSUP when CONFIG_FOTA_DOWNLOAD is not enabled");
+}
+
+/* Verify that nrf_cloud_bootloader_fota_slot_set returns success with no flag changes
+ * for the following conditions:
+ *   When a non-BOOT FOTA job is provided.
+ *   When a BOOT FOTA job is provided with the s0 set flag already set.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_bootloader_fota_slot_success_no_change)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_FOTA_DOWNLOAD);
+	struct nrf_cloud_settings_fota_job job;
+
+	/* Verify no flag changes when a non-BOOT FOTA type is provided */
+	job.bl_flags = NRF_CLOUD_FOTA_BL_STATUS_CLEAR;
+	job.type = NRF_CLOUD_FOTA_APPLICATION;
+
+	int ret = nrf_cloud_bootloader_fota_slot_set(&job);
+
+	zassert_equal(0, ret,
+		      "return should be 0 when not a BOOT type");
+	zassert_equal(NRF_CLOUD_FOTA_BL_STATUS_CLEAR, job.bl_flags,
+		      "bl_flags should be NRF_CLOUD_FOTA_BL_STATUS_CLEAR");
+
+	/* Verify no flag changes when the s0 set flag is already set for a BOOT FOTA type */
+	job.bl_flags = NRF_CLOUD_FOTA_BL_STATUS_S0_FLAG_SET;
+	job.type = NRF_CLOUD_FOTA_BOOTLOADER;
+
+	ret = nrf_cloud_bootloader_fota_slot_set(&job);
+
+	zassert_equal(0, ret,
+		      "return should be 0 when flag already set");
+	zassert_equal(NRF_CLOUD_FOTA_BL_STATUS_S0_FLAG_SET, job.bl_flags,
+		      "bl_flags should be NRF_CLOUD_FOTA_BL_STATUS_S0_FLAG_SET");
+}
+
+/* Verify that nrf_cloud_bootloader_fota_slot_set returns success and the flags are set correctly
+ * when a BOOT FOTA job is provided with the bl_flags cleared.
+ * This test checks both states of s0 active.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_bootloader_fota_slot_success)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_FOTA_DOWNLOAD);
+	struct nrf_cloud_settings_fota_job job;
+
+	/* Test for s0 active */
+	job.bl_flags = NRF_CLOUD_FOTA_BL_STATUS_CLEAR;
+	job.type = NRF_CLOUD_FOTA_BOOTLOADER;
+	fota_download_s0_active_get_fake.custom_fake = fake_fota_download_s0_active_get__s0_active;
+
+	int ret = nrf_cloud_bootloader_fota_slot_set(&job);
+
+	zassert_equal(0, ret,
+		      "return should be 0");
+	zassert_equal(NRF_CLOUD_FOTA_BL_STATUS_S0_FLAG_SET|NRF_CLOUD_FOTA_BL_STATUS_S0_WAS_ACTIVE,
+		      job.bl_flags,
+		      "bl_flags should indicate s0 set and s0 active");
+
+	/* Test for s0 not active (s1 active) */
+	job.bl_flags = NRF_CLOUD_FOTA_BL_STATUS_S0_WAS_ACTIVE;
+	fota_download_s0_active_get_fake.custom_fake =
+		fake_fota_download_s0_active_get__s0_inactive;
+
+	ret = nrf_cloud_bootloader_fota_slot_set(&job);
+
+	zassert_equal(0, ret,
+		      "return should be 0");
+	zassert_equal(NRF_CLOUD_FOTA_BL_STATUS_S0_FLAG_SET, job.bl_flags,
+		      "bl_flags should indicate s0 set and s0 inactive");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process fails for an unknown FOTA job type. */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_unknwn_type)
```

Comment:
is this intended or typo? `test_nrf_cloud_pending_fota_job_process_**unknwn**_type`?

Reply:
probably intended, but not necessary... i'll add the `o` 😅